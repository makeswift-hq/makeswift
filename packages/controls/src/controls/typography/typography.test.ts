import {
  unstable_Typography,
  unstable_TypographyDefinition,
} from './typography'

describe('Typography', () => {
  describe('constructor', () => {
    test('returns correct definition', () => {
      expect(unstable_Typography()).toMatchSnapshot()
    })

    test('disallows extraneous properties', () => {
      // @ts-expect-error
      unstable_Typography({
        extra: 'extra',
      })
    })
  })

  describe('assignability', () => {
    function assignTest(_def: unstable_TypographyDefinition) {}
    assignTest(unstable_Typography())
  })
})
