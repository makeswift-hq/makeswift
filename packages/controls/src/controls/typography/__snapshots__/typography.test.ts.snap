// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography constructor returns correct definition 1`] = `
unstable_TypographyDefinition {
  "config": {},
}
`;
